### DISCUSSION TITLE
How to manipulate IFC tree data using IFCOpenshell using Python

### DISCUSSION THREAD

**Discussion by Akshay3241** [IfcOpenShell/IfcOpenShell#4458] [General]
Hello everyone,

Here I am trying to move IFCElement from one IFCBuildingStory to another IFCBuildingStory.
Is this possible?
I have written small code for this as well. I have tried IFCRelSpatialContainedStructure here.

ifc_file_path = "C:\\Users\\sabaleak\\Downloads\\P06_Test.ifc"
>>> ifc_file = ifcopenshell.open(ifc_file_path)
>>> element_id = 34296
>>> element = ifc_file.by_id(element_id)
>>> source_storey_id = 78
>>> source_storey = ifc_file.by_id(source_storey_id)
>>> target_storey_id = 71
>>> target_storey = ifc_file.by_id(target_storey_id)
>>> source_relationships = ifc_file.by_type("IfcRelContainedInSpatialStructure")
>>> source_relationship = None
>>> for rel in source_relationships:
...     if rel.RelatingStructure.id() == source_storey.id():
...         source_relationship = rel
...         break
...
>>> related_elements = source_relationship.RelatedElements
>>> related_elements =list(source_relationship.RelatedElements)
>>> related_elements = [elem for elem in related_elements if elem.id() != 34296]
>>> related_elements_tuple=tuple(related_elements)
>>> source_relationship=related_elements_tuple
>>> ifc_file.write("C:\\Users\\sabaleak\\Downloads\\P06_Test_Modified.ifc")

>>> target_relationship = ifc_file.create_entity("IfcRelContainedInSpatialStructure")
>>> target_relationship.RelatingStructure = target_storey
>>> w= list(target_relationship)
>>> w.append(element)
>>> target_relationship=tuple(w)
>>> ifc_file.write("C:\\Users\\sabaleak\\Downloads\\P06_Test_Modified.ifc")

Here I want to move #34296 element from one buildingStory to another.
I have user target and source.

Anyone leads would be appritiated.

**Comment by aothms** [comment]
These days there is a high level API for these kind of tasks. Have a look at: https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/ifcopenshell-python/ifcopenshell/api/spatial/assign_container.py

**Reply by Akshay3241** [reply]
Sure, I will check this.
Thanks for your help
